### PR TITLE
feat(api): add basic model types

### PR DESCRIPTION
**Pull request by iartemiev**
#### Description of changes

Simplified version of https://github.com/aws-amplify/amplify-js/pull/11671 without template literal types.

* Adds new V6 client types.

#### Checklist

- [x] PR description included
- [x] `yarn test` passes
- [ ] Tests are [changed or added](https://github.com/aws-amplify/amplify-js/blob/main/CONTRIBUTING.md#steps-towards-contributions)
- [ ] Relevant documentation is changed or added (and PR referenced)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/api/src/API.ts [hunk 1]
```diff
@@ -67,14 +67,22 @@ export class APIClass extends InternalAPIClass {
...
+					client.models[name][operationPrefix] = async (arg?: any) => {
 						const query = generateGraphQLDocument(model, operation);
 						const variables = buildGraphQLVariables(model, operation, arg);
 
-						return this.graphql({
+						const res = (await this.graphql({
 							query,
 							variables,
-						});
+						})) as any;
```
Comment (svidgen) [inline packages/api/src/API.ts:77]: Could we benefit from some validation and/or type guarding here?

File: packages/api/src/API.ts [hunk 2]
```diff
@@ -252,13 +260,31 @@ type ExcludeNeverFields<O> = {
 	[K in FilteredKeys<O>]: O[K];
 };
 
+type Prettify<T> = T extends object
+	? {
+			[P in keyof T]: Prettify<T[P]>;
+	  }
+	: T;
```
Comment (svidgen) [inline packages/api/src/API.ts:267]: What's this do?

**Review by svidgen — APPROVED** [review]

File: packages/api/src/API.ts (hunk 2 above)
Comment (iartemiev) [reply packages/api/src/API.ts:267]: Makes complex/nested/union/intersection types human-readable. E.g. 
`{a: string} & {b: number}` => `{a: string; b: number}`

Reference: https://twitter.com/mattpocockuk/status/1622730179289251840?s=20

File: packages/api/src/API.ts (hunk 1 above)
Comment (iartemiev) [reply packages/api/src/API.ts:77]: Yes, this is just temporary. Will add a type guard that checks whether it's `GraphQLResult` or `Observable`

**Pull request merged**